pkgs/development/python-modules/torch/test-cuda.nix

```diff
@@ -0,0 +1,22 @@
+{ runCommandNoCC
+, python
+, torch
+}:
+
+runCommandNoCC "pynvml-gpu-test"
+{
+  nativeBuildInputs = [
+    (python.withPackages (_: [ torch ]))
+  ];
```

Comment:
Seeing as we can use ``runCommand`` with this, it also seems to be possible to script together a test that can have acceptable failure conditions.
This is important to me because ``rsmitst`` from ``rocm-smi`` requires root, which we obviously do not want to give. I did test out a root-allowing variant with ``makeImpureTest``, but I don't prefer it.
I think the approach with this PR is superior, and more streamlined.

Reply:
That would be ideal, since we could just mock root in the VM and not actually have to give the nix builder real root access to the host.